internal/grpcutil/target_test.go

```diff
@@ -91,6 +91,8 @@ func TestParseTargetString(t *testing.T) {
 		{targetStr: "unix-abstract:/ a///://::!@#$%^&*()b", want: resolver.Target{Scheme: "unix-abstract", Authority: "", Endpoint: "/ a///://::!@#$%^&*()b"}},
 		{targetStr: "unix-abstract:passthrough:abc", want: resolver.Target{Scheme: "unix-abstract", Authority: "", Endpoint: "passthrough:abc"}},
 		{targetStr: "unix-abstract:unix:///abc", want: resolver.Target{Scheme: "unix-abstract", Authority: "", Endpoint: "unix:///abc"}},
+		{targetStr: "unix-abstract:///a/b/c", want: resolver.Target{Scheme: "unix-abstract", Authority: "", Endpoint: "///a/b/c"}},
+		{targetStr: "unix-abstract://authority/a/b/c", want: resolver.Target{Scheme: "unix-abstract", Authority: "", Endpoint: "//authority/a/b/c"}},
```

Comment:
This target string *should* parse as `Authority: "authority", Endpoint: "/a/b/c"`

Reply:
The `unix-abstract` schema has no authority by the definition in https://github.com/grpc/grpc/blob/master/doc/naming.md, same as `unix` schema.

Am I missing something?

Reply:
Theoretically, *all* schemes should use the exact same logic to parse a URI into its components.  We unfortunately have some hackery around here for backward compatibility reasons.

It's illegal for an authority to be specified for `unix-abstract`, but that should make the name resolver fail the attempt to include one, not parse it as part of the endpoint.

Reply:
Understood, if this is for backward compatibility reasons, I am fine with it.

Just one final confirmation, if we do parse the authority, we are actually compromising the namespace of `unix-abstract`? Because it is legal by definition that user can use any bytes as the abstract socket name, yet our parser blocks it.

Reply:
> Understood, if this is for backward compatibility reasons, I am fine with it.

Not quite.  There is some hackery in parsing existing schemes (e.g. `unix`) for backward compatibility.  I want `unix-abstract` to be parsed the way all target strings are supposed to be parsed.

> Just one final confirmation, if we do parse the authority, we are actually compromising the namespace of unix-abstract? Because it is legal by definition that user can use any bytes as the abstract socket name, yet our parser blocks it.

We aren't compromising anything; this is how Java and C do their parsing here as well.  The socket name *can* be any sequence of bytes, but they need to be specified correctly.  I.e. if you want the endpoint to be "//a/b/c" for some reason, then you will need to use `"unix-abstract://///a/b/c"` because we don't currently support percent-encoding; otherwise you could do `"unix-abstract:%2F%2Fa%2Fb%2Fc"` to avoid making the `"//a/"` appear like it's specifying an authority of `a` and an endpoint of "/b/c".

Basically we really want RFC 3986-compliant URI parsing, but we have to grandfather in uses from when our parsing was ad-hoc and wrong in some cases.  For new schemes, we probably could use `url.Parse` but let's push off such a major change for another time.

Reply:
Understood, thanks for your detailed explanation. Already added the code for authority parsing for `unix-abstract` schema

Also added two more test cases for `unix-abstract` schema:
- `unix-abstract:///` parsed as empty Authority and Endpoint `/`
- `unix-abstract://authority` parsed as empty Authority and Endpoint `//authority`

Let me know if they are unexpected. Thanks for your time.

Reply:
> Also added two more test cases for unix-abstract schema:

Those also sound like great test cases, thanks.